### PR TITLE
feature(notice): analytics consent notice

### PR DESCRIPTION
**Pull request by jorbuedo**
Ticket: https://emurgo.atlassian.net/browse/YOMO-659

Adds the consent notice for analytics. Using a Modal for now, once BottomSheet is merged in it can be easily changed.

image

### CODE REVIEW

File: apps/wallet-mobile/src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx
Comment (stackchain) [inline apps/wallet-mobile/src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx:1]: can we have the story with consent open?

File: apps/wallet-mobile/src/legacy/DeveloperScreen.tsx [hunk 1]
```diff
@@ -109,6 +112,14 @@ export const DeveloperScreen = () => {
           }}
         />
 
+        <Button
+          title={`Reset Analytics Consent (${isConsentRequested})`}
```
Comment (banklesss) [inline apps/wallet-mobile/src/legacy/DeveloperScreen.tsx:116]: intl?

Comment (banklesss) [reply apps/wallet-mobile/src/legacy/DeveloperScreen.tsx:116]: ah ok developer screen !

File: apps/wallet-mobile/src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx
Comment (jorbuedo) [reply apps/wallet-mobile/src/SelectedWallet/WalletSelection/WalletSelectionScreen.tsx:1]: It is open on storybook, but I'm not 100% sure how the mocked storage is dealing with it.

File: apps/wallet-mobile/src/legacy/DeveloperScreen.tsx (hunk 1 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/legacy/DeveloperScreen.tsx:116]: That's the developer screen, no translations there.

**Pull request merged**